src/libraries/System.Net.Security/src/System/Net/Security/ReadWriteAdapter.cs

```diff
@@ -40,9 +40,9 @@ internal interface IReadWriteAdapter
         public static ValueTask<int> ReadAtLeastAsync(Stream stream, Memory<byte> buffer, int minimumBytes, bool throwOnEndOfStream, CancellationToken cancellationToken) =>
             new ValueTask<int>(stream.ReadAtLeast(buffer.Span, minimumBytes, throwOnEndOfStream));
 
-        public static ValueTask WriteAsync(Stream stream, byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        public static ValueTask WriteAsync(Stream stream, ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
         {
-            stream.Write(buffer, offset, count);
+            stream.Write(buffer.Span);
```

Comment:
I realise that calling this overload could have a potential negative performance impact on `Stream`s that don't implement `Write(ReadOnlySpan<byte>)`. Firstly, the `Read(Span<byte>)` overload is already used so this only aligns the usage. Secondly, it can skip couple of parameter checks. Lastly, `SslStream`/`NegotiateStream` are usually layered on top of transport streams, such as `NetworkStream`, that do implement the `Write(Span<byte>)` overload.

Reply:
Yup, I think it's fine.